watcher/src/watcher.rs

```diff
@@ -194,12 +195,14 @@ impl Watcher {
                     }
 
                     Err(err) => {
-                        log::debug!(
-                            self.logger,
-                            "Could not sync block {} for url ({:?})",
-                            block_index,
-                            err
-                        );
+                        if log_sync_failures {
```

Comment:
I think this should `else` into a debug log, otherwise this could potentially degrade debugging experience when this flag is set to false.

Reply:
Agreed